crml/cennzx-spot/src/lib.rs

```diff
@@ -55,8 +56,83 @@ pub trait Trait: frame_system::Trait + pallet_generic_asset::Trait {
 	type UnsignedIntToBalance: From<LowPrecisionUnsigned> + Into<<Self as pallet_generic_asset::Trait>::Balance>;
 }
 
+decl_error! {
+	/// Error for the generic-asset module.
+	pub enum Error for Module<T: Trait> {
+		/// Exchange pool is empty.
+		EmptyExchangePool,
+		// Insufficient trade asset reserve in exchange
+		InsufficientTradeAssetReserve,
+		// Insufficient core asset reserve in exchange
+		InsufficientCoreAssetReserve,
+		// Insufficient asset balance in buyer account
+		InsufficientBuyerTradeAssetBalance,
```

Comment:
I think we probably don't need that many error messages as the type of errors is the same in many places. Similar to the Overflow error that we have used in many places regardless of where it has happened, we could have a few error types such as InsufficientAssetBalance, AssetNotAboveZero, AmountNotAboveZero,... 
However, as this is a refactor which is not pursuing any significant output change, I can see you may not want to go down that route.

Reply:
Yeah, there are as many error types as there were unique messages in the past. Totally agree that we should consider reducing the number of errors going forward